EvolutionModel/Foodbag.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Evolution.Model {
+    public class Foodbag {
+        public uint Foodamount {
+            get;
+            private set;
+        }
+
+        public Foodbag(){ 
+            Foodamount = 0;
+        }
+
+        public uint Count => Foodamount;
+
+        public void AddFood(uint erbij){
+            Foodamount += erbij;
```

Comment:
Technisch gezien zouden we hier in de problemen kunnen komen als `Foodamount = uint.MaxValue` en `erbij = uint.MaxValue`. Dan hebben we namelijk een integer overflow. Je kan aan de compiler laten weten dat expliciet te voorkomen door middel van een `checked` blok:

```
public int AddLarge(int x) {
  return x + int.MaxValue;
}
public int AddLargeSafe(int x) {
  checked {
    return x + int.MaxValue;
  }
}
> AddLarge(5)
-2147483644

AddLargeSafe(5);
Arithmetic operation resulted in an overflow.
```

In het tweede geval gooit hij dan een `OverflowException`.

Reply:
Hier kan je vervolgens ook op testen :)

Reply:
Leg dit vanavond nog maar een keer uit ;)

Reply:
Kijk [hier](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/checked) anders nog even.